### ISSUE TITLE
Make kubelet-cri-gce-e2e-ci/gci-gce-cri green

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#33189]
Metaissue: #31459

@Random-Liu set up a new builder to test the new docker CRI integration.
node e2e: https://k8s-testgrid.appspot.com/google-node#kubelet-cri-gce-e2e
cluster e2e: https://k8s-testgrid.appspot.com/google-gce#gci-gce-cri

Right now, it runs non-serial node e2e tests, with 27 tests passing.
`FAIL! -- 27 Passed | 75 Failed | 0 Pending | 37 Skipped`

We need to make the build green.

/cc @kubernetes/sig-node 

Known issues: exec/logs are not supported yet in the CRI implementation, pending more discussions on the API. We bypass CRI to implement logs/exec for now to unblock other tests, since those two operations are commonly used by tests to verify results.

**Comment by yujuhong** [comment]
The lack of `exec` and `logs` are causing many test failures because tests often rely on these two operations to verify the output. To unblock work on the integration while continue addressing `exec` and `logs`, I'd submit a PR to hack around that (i.e., using the docker exec/logs directly) today.

**Comment by yujuhong** [comment]
After faking `exec` and `logs`, 94 tests passed and 8 failed. 

The failed tests are
 - [ ] **4 init container tests**
 - [ ] **apparmor test**
 - [ ] **Kubelet metrics api when querying /stats/summary it should report resource usage through the stats api.**
 - [ ] **Kubelet when scheduling a busybox command that always fails in a pod should have an error terminated reason** 
 - [ ] **Pods should have their auto-restart back-off timer reset on image update [Slow]**

@feiskyer is working on adding init container support in kuberuntime. 
I'll take a look at the "terminated reason" test.

**Comment by dchen1107** [comment]
Cool, such great progress!

**Comment by yujuhong** [comment]
> I'll take a look at the "terminated reason" test.

I uploaded #33386 to fix this test. However, the test won't be valid any more since we don't control what "Reason" the runtime returns.

**Comment by Random-Liu** [comment]
I'll take a look at the metrics api one.

**Comment by Random-Liu** [comment]
All right, the metrics api test fails just because image fs info is empty.

**Comment by Random-Liu** [comment]
If so, I'll take a look at the backoff test.

**Comment by Random-Liu** [comment]
Except those failed test, there are also some flaky ones:
- [ ] Container Runtime Conformance Test container runtime conformance blackbox test when starting a container that exits should report termination message if TerminationMessagePath is set
- [ ] Downward API should provide default limits.cpu/memory from node allocatable
- [ ] Downward API should provide pod name and namespace as env vars
- [ ] HostPath should give a volume the correct mode
- [ ] Variable Expansion should allow composing env vars into new env vars

**Comment by yifan-gu** [comment]
> After faking exec and logs, 94 tests passed and 8 failed.

GOOD JOB!!

**Comment by Random-Liu** [comment]
FYI, here is a cri e2e test flake: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/259

**Comment by Random-Liu** [comment]
@yujuhong kubelet-cri-gce-e2e-ci is green now. :)

**Comment by yujuhong** [comment]
Great! Let's close this one and start looking at serial/soak tests.

**Comment by yujuhong** [comment]
> @yujuhong Cluster e2e still needs some love, but node e2e is all green now.

Last I checked it was all green?... https://k8s-testgrid.appspot.com/google-gce#gci-gce-cri

**Comment by Random-Liu** [comment]
@yujuhong There are several flakes. Till now, all flakes turned out to be a bug. :)

**Comment by yujuhong** [comment]
> @yujuhong There are several flakes. Till now, all flakes turned out to be bugs. :)

I am actually quite happy that flakes are caused by bugs (the tests are doing something!). Let's reuse this issue to track the current flakes in the cluster e2e tests.

**Comment by Random-Liu** [comment]
I checked one e2e test failure and found it failed because pod sandbox was not started:
```
Oct  6 09:56:28.866: INFO: At 2016-10-06 09:51:27 -0700 PDT - event for pod-1fd4e7e9-8be5-11e6-b5ec-0242ac110002: {default-scheduler } Scheduled: Successfully assigned pod-1fd4e7e9-8be5-11e6-b5ec-0242ac110002 to jenkins-e2e-minion-group-f6m7
Oct  6 09:56:28.866: INFO: At 2016-10-06 09:51:28 -0700 PDT - event for pod-1fd4e7e9-8be5-11e6-b5ec-0242ac110002: {kubelet jenkins-e2e-minion-group-f6m7} SandboxReceived: Pod sandbox received, it will be created.
Oct  6 09:56:28.866: INFO: At 2016-10-06 09:51:29 -0700 PDT - event for pod-1fd4e7e9-8be5-11e6-b5ec-0242ac110002: {kubelet jenkins-e2e-minion-group-f6m7} FailedSync: Error syncing pod, skipping: failed to "SetupNetwork" for "929e2bb7d96f906ee9cbef6b477a7a592c217ffdfc615856437b0044d950ec50" with SetupNetworkError: "Failed to setup network for pod \"pod-1fd4e7e9-8be5-11e6-b5ec-0242ac110002_e2e-tests-emptydir-tcd2r(1fdaf9c3-8be5-11e6-bd78-42010af00002)\" using network plugins \"kubenet\": Failed to execute iptables-restore: exit status 1 (iptables-restore: line 3 failed\n)"

Oct  6 09:56:28.866: INFO: At 2016-10-06 09:51:30 -0700 PDT - event for pod-1fd4e7e9-8be5-11e6-b5ec-0242ac110002: {kubelet jenkins-e2e-minion-group-f6m7} Pulled: Container image "gcr.io/google_containers/mounttest:0.7" already present on machine
Oct  6 09:56:28.866: INFO: At 2016-10-06 09:51:30 -0700 PDT - event for pod-1fd4e7e9-8be5-11e6-b5ec-0242ac110002: {kubelet jenkins-e2e-minion-group-f6m7} Created: Created container with id c26d9dc574dde55bb016f13062b3310bd8398a063f4dab0962aa4cf274f8f043
Oct  6 09:56:28.866: INFO: At 2016-10-06 09:51:30 -0700 PDT - event for pod-1fd4e7e9-8be5-11e6-b5ec-0242ac110002: {kubelet jenkins-e2e-minion-group-f6m7} Failed: Failed to start container with id c26d9dc574dde55bb016f13062b3310bd8398a063f4dab0962aa4cf274f8f043 with error: Error response from daemon: cannot join network of a non running container: 929e2bb7d96f906ee9cbef6b477a7a592c217ffdfc615856437b0044d950ec50
```

`\"kubenet\": Failed to execute iptables-restore: exit status 1 (iptables-restore: line 3 failed\n)"` @freehan

**Comment by Random-Liu** [comment]
Another flake `Job should scale a job up 5m20s`. Will investigate it.

**Comment by Random-Liu** [comment]
I checked these 2 days result of gci-gce-cri. Here are 4 kinds of failures I've found:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/565
```
Couldn't delete ns: "e2e-tests-services-3zub2": namespace e2e-tests-services-3zub2 was not deleted with limit: timed out waiting for the condition, pods remaining: 1, pods missing deletion timestamp: 0
```

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/576
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/561
```
Error syncing pod, skipping: failed to "SetupNetwork" for "3ffd1b548478b373ede9f188e9ae9c78bf0eaab63ba64571c198d47150b2c16a" with SetupNetworkError: "Failed to setup network for pod \"pod2_e2e-tests-services-cla4f(0fde4343-8fa7-11e6-8c79-42010af00002)\" using network plugins \"kubenet\": Failed to ensure that nat chain POSTROUTING jumps to MASQUERADE: error checking rule: exit status 4: iptables: Resource temporarily unavailable.\n"
```

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/549
```
FailedSync: Error syncing pod, skipping: failed to "CreatePodSandbox" for "rs-fi5no_e2e-tests-disruption-bj1jl(33e0cbe0-8f37-11e6-ba25-42010af00002)" with CreatePodSandboxError: "CreatePodSandbox for pod \"rs-fi5no_e2e-tests-disruption-bj1jl(33e0cbe0-8f37-11e6-ba25-42010af00002)\" failed: failed to create a sandbox for pod \"rs-fi5no\": Error response from daemon: Conflict. The name \"/k8s_POD_rs-fi5no_e2e-tests-disruption-bj1jl_33e0cbe0-8f37-11e6-ba25-42010af00002_0\" is already in use by container 21f37ad6319bdb09f97b477c2b8e3c3d5fb01014ab2fb04e63417e751e44347e. You have to remove (or rename) that container to be able to reuse that name."
```


https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/543
```
{kubelet jenkins-e2e-minion-group-dxon} FailedSync: Error syncing pod, skipping: failed to "SetupNetwork" for "77eb3c4db2261a3c4229df34901f861e1a3600ce82be29080315929ceb322a59" with SetupNetworkError: "Failed to setup network for pod \"my-hostname-private-aee772f6-8f1a-11e6-947d-0242ac11000a-izeto_e2e-tests-replicaset-ran80(aef25d2a-8f1a-11e6-921d-42010af00002)\" using network plugins \"kubenet\": Failed to execute iptables-restore: exit status 1 (iptables-restore: line 3 failed\n)"
```

**Comment by Random-Liu** [comment]
A new test introduced is failing
```
Cluster level logging using GCL should check that logs from containers are ingested in GCL
```

**Comment by yujuhong** [comment]
> Cluster level logging using GCL should check that logs from containers are ingested in GCL

Uh...there are some issues with our implementation, but the test is fundamentally wrong by not setting restart policy to "never"

**Comment by Random-Liu** [comment]
> Uh...there are some issues with our implementation, but the test is fundamentally wrong by not setting restart policy to "never"

Hm...The test is all green in other suites. Why does it success?

**Comment by Random-Liu** [comment]
OK. That is because the function [`waitForPodSuccessInNamespaceTimeout`](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1468-L1487) is also fundamentally wrong.

We discussed this before, but I didn't realize the current logic is wrong.
`(Container Termianted with ExitCode == 0) != (Pod Success)`

I'll send a PR to fix the test and the framework, and fix the dockershim to do symlink for now.

**Comment by feiskyer** [comment]
Seems #34669 has fixed network problems. Only one failed now:

```
[k8s.io] Cluster level logging using GCL should check that logs from containers are ingested in GCL 10m22s
```

**Comment by Random-Liu** [comment]
> Seems #34669 has fixed network problems. Only one failed now:

Seems not. https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/855
```
event for rs-qrcy0: {kubelet jenkins-e2e-minion-group-cvwe} FailedSync: Error syncing pod, skipping: failed to "SetupNetwork" for "964bb9384d6c0e7098ded81c46efeb05294786d6779b1a0f25a9b59885099e56" with SetupNetworkError: "Failed to setup network for pod \"rs-qrcy0_e2e-tests-disruption-jubfy(55a052f3-9438-11e6-a9f4-42010af00002)\" using network plugins \"kubenet\": Failed to ensure that nat chain POSTROUTING jumps to MASQUERADE: error checking rule: exit status 4: iptables: Resource temporarily unavailable.\n"
```

**Comment by yujuhong** [comment]
Uh...is this error unique to the CRI suite? "Resource temporarily unavailable" seems generic enough that it could've happened in other integration too...

**Comment by Random-Liu** [comment]
I found this in the test `DisruptionController evictions: enough pods, replicaSet, percentage => should allow an eviction`, which is all green in `gci-gce`.

**Comment by Random-Liu** [comment]
A new issue we may want to keep an eye one https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubelet-cri-gce-e2e-ci/502. Some observations:
* Docker is down when the test failed (around 00:35:00).
* From that time on, there is no more [kern.log](https://storage.googleapis.com/kubernetes-jenkins/logs/kubelet-cri-gce-e2e-ci/502/artifacts/tmp-node-e2e-b78c3331-gci-dev-55-8820-0-0/kern.log) or [docker.log](https://storage.googleapis.com/kubernetes-jenkins/logs/kubelet-cri-gce-e2e-ci/502/artifacts/tmp-node-e2e-b78c3331-gci-dev-55-8820-0-0/docker.log)
* Kubelet is still running and logging [kubelet.log](https://storage.googleapis.com/kubernetes-jenkins/logs/kubelet-cri-gce-e2e-ci/502/artifacts/tmp-node-e2e-b78c3331-gci-dev-55-8820-0-0/kubelet.log)

**Comment by Random-Liu** [comment]
Another flake in gci-gce-cri, which I think @yujuhong mentioned to me last week.
* Link: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/1157
* Failure: 
```
Port forwarding [k8s.io] With a server that expects a client request should support a client that connects, sends no data, and disconnects [Conformance] 5m27s
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/portforward.go:220
Oct 24 05:05:50.313: Pod did not start running: timed out waiting for the condition
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/portforward.go:184
```
* Symptoms:
* dockershim failed to create infra container with error `device or resource busy`, but when it tried to create infra container again later, docker claimed that the container already existed. (See https://storage.googleapis.com/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/1157/build-log.txt)
```
Oct 24 05:05:50.347: INFO: At 2016-10-24 05:00:52 -0700 PDT - event for pfpod: {kubelet jenkins-e2e-minion-group-89b1} FailedSync: Error syncing pod, skipping: failed to "CreatePodSandbox" for "pfpod_e2e-tests-port-forwarding-dlxt2(81a3469e-99e1-11e6-89f2-42010af00002)" with CreatePodSandboxError: "CreatePodSandbox for pod \"pfpod_e2e-tests-port-forwarding-dlxt2(81a3469e-99e1-11e6-89f2-42010af00002)\" failed: failed to create a sandbox for pod \"pfpod\": Error response from daemon: device or resource busy"
Oct 24 05:05:50.347: INFO: At 2016-10-24 05:01:07 -0700 PDT - event for pfpod: {kubelet jenkins-e2e-minion-group-89b1} FailedSync: Error syncing pod, skipping: failed to "CreatePodSandbox" for "pfpod_e2e-tests-port-forwarding-dlxt2(81a3469e-99e1-11e6-89f2-42010af00002)" with CreatePodSandboxError: "CreatePodSandbox for pod \"pfpod_e2e-tests-port-forwarding-dlxt2(81a3469e-99e1-11e6-89f2-42010af00002)\" failed: failed to create a sandbox for pod \"pfpod\": Error response from daemon: Conflict. The name \"/k8s_POD_pfpod_e2e-tests-port-forwarding-dlxt2_81a3469e-99e1-11e6-89f2-42010af00002_0\" is already in use by container 24666ab8c814d16f986449e504ea0159468ddf8da01897144a770f66dce0e14e. You have to remove (or rename) that container to be able to reuse that name."
```
* docker failed to create the container because failed to unmount overlay: (See https://storage.googleapis.com/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/1157/artifacts/jenkins-e2e-minion-group-89b1/docker.log)
```
time="2016-10-24T12:00:52.611940107Z" level=debug msg="Start untar layer"
time="2016-10-24T12:00:52.614355610Z" level=debug msg="Failed to unmount 0725dd0edc1d9ebe4e80040f7fd110ade6544683ce12e639438578e399321729-init overlay: device or resource busy"
time="2016-10-24T12:00:52.614459785Z" level=error msg="Clean up Error! Cannot destroy container 24666ab8c814d16f986449e504ea0159468ddf8da01897144a770f66dce0e14e: No such container: 24666ab8c814d16f986449e504ea0159468ddf8da01897144a770f66dce0e14e"
```
* kernel log showed the kernel race `
unregister_netdevice: waiting for lo to become free. Usage count = 1`: (See https://storage.googleapis.com/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/1157/artifacts/jenkins-e2e-minion-group-89b1/kern.log)
```
device vethba199a46 left promiscuous mode
cbr0: port 4(vethba199a46) entered disabled state
cbr0: port 2(veth2f59c2d5) entered disabled state
device veth2f59c2d5 left promiscuous mode
cbr0: port 2(veth2f59c2d5) entered disabled state
cbr0: port 5(veth8722e769) entered disabled state
device veth8722e769 left promiscuous mode
cbr0: port 5(veth8722e769) entered disabled state
unregister_netdevice: waiting for lo to become free. Usage count = 1
```

@yujuhong @feiskyer /cc @bprashanth

**Comment by bprashanth** [comment]
Please link kern.log if you get a chance. A single instance of that message (unregistered_netdevice) is probably not the same as the old bug, but it's possible that it caused some disruption. If you see O(ten) instances spamming the log, and a docker hung task/the node completely dead, that's probably the dreaded kernel reace we mitigated by making cbr0 promiscuous.

**Comment by yujuhong** [comment]
A similar failure I mentioned to @Random-Liu previously:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/35245/pull-kubernetes-e2e-gce-cri/12/
The node in question: http://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/35245/pull-kubernetes-e2e-gce-cri/12/artifacts/cri-e2e-agent-pr-25-0-minion-group-ilqm/

Saw similar "failed to unmount" error in docker.log;
```
time="2016-10-21T04:41:30.282650176Z" level=debug msg="Failed to unmount 7e418df02da3d8e4695c5fe45ace30823d7164eb47d5705748125b455466ec7d-init overlay: device or resource busy"
time="2016-10-21T04:41:30.282911615Z" level=error msg="Clean up Error! Cannot destroy container ad2387bd44e9ad568776d216de0c1b76720aee784c18f24b7eb65d6126181964: No such container: ad2387bd44e9ad568776d216de0c1b76720aee784c18f24b7eb65d6126181964"
```

Didn't see the unregister_netdevice error in the kernel.log. Perhaps a docker+overlayfs bug?

**Comment by Random-Liu** [comment]
> Perhaps a docker+overlayfs bug?

However, I don't remember we saw this anywhere in non-cri integration.

**Comment by yujuhong** [comment]
> However, I don't remember we saw this anywhere in non-cri integration.

We wouldn't notice this at all if kubelet was able to create a new infra container with a randomly generated suffix in the name, which is what we had been doing pre-CRI.

**Comment by Random-Liu** [comment]
> We wouldn't notice this at all if kubelet was able to create a new infra container with a randomly generated suffix in the name, which is what we had been doing pre-CRI.

@yujuhong Good point! You mentioned this before, forgot that. :)

**Comment by Random-Liu** [comment]
Two failure recently:

1) e2e-image-puller was terminated by someone. https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/1961
I didn't find anything weird in kubelet.log, but the container failed with a `Terminated` message. It seems that someone killed it, but not kubelet, because we can't see [this log](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/kuberuntime_container.go#L509).
```
19:09:13 pulling gcr.io/google_containers/busybox
19:09:13 pulling gcr.io/google_containers/busybox:1.24
19:09:14 pulling gcr.io/google_containers/dnsutils:e2e
19:09:20 pulling gcr.io/google_containers/eptest:0.1
19:09:20 pulling gcr.io/google_containers/fakegitserver:0.1
19:09:21 pulling gcr.io/google_containers/hostexec:1.2
19:09:22 pulling gcr.io/google_containers/iperf:e2e
19:09:25 pulling gcr.io/google_containers/jessie-dnsutils:e2e
19:09:32 pulling gcr.io/google_containers/liveness:e2e
19:09:33 pulling gcr.io/google_containers/mounttest:0.7
19:09:33 pulling gcr.io/google_containers/mounttest-user:0.3
19:09:33 pulling gcr.io/google_containers/netexec:1.4
19:09:34 pulling gcr.io/google_containers/netexec:1.7
19:09:35 pulling gcr.io/google_containers/nettest:1.7
19:09:36 pulling gcr.io/google_containers/nettest:1.8
19:09:37 pulling gcr.io/google_containers/nginx-slim:0.7
19:09:42 pulling gcr.io/google_containers/n-way-http:1.0
19:09:43 pulling gcr.io/google_containers/pause:2.0
19:09:43 pulling gcr.io/google_containers/pause-amd64:3.0
19:09:43 pulling gcr.io/google_containers/porter:cd5cb5791ebaa8641955f0e8c2a9bed669b1eaab
19:09:44 pulling gcr.io/google_containers/portforwardtester:1.0
19:09:45 pulling gcr.io/google_containers/redis:e2e
Terminated
19:10:27 pulling gcr.io/google_containers/resource_consumer:beta4
```

2) NPD failed to start because of `device or resource busy`. https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/1961
This is a known issue, we should work around this.
```
I1109 21:43:29.623032    1369 server.go:664] Event(api.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"node-problem-detector-v0.1-w9ktn", UID:"8a66a07a-a6c5-11e6-8ace-42010af00002", APIVersion:"v1", ResourceVersion:"144", FieldPath:"spec.containers{node-problem-detector}"}): type: 'Warning' reason: 'Failed' Failed to create container with error: rpc error: code = 2 desc = Error response from daemon: device or resource busy
I1109 21:43:30.707743    1369 server.go:664] Event(api.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"node-problem-detector-v0.1-w9ktn", UID:"8a66a07a-a6c5-11e6-8ace-42010af00002", APIVersion:"v1", ResourceVersion:"144", FieldPath:"spec.containers{node-problem-detector}"}): type: 'Warning' reason: 'Failed' Failed to create container with error: rpc error: code = 2 desc = Error response from daemon: Conflict. The name "/k8s_node-problem-detector_node-problem-detector-v0.1-w9ktn_kube-system_8a66a07a-a6c5-11e6-8ace-42010af00002_0" is already in use by container 008291f32f1fea534e691a38b6ddc7021a3091e36ed7a96db87e1fd46d16bf36. You have to remove (or rename) that container to be able to reuse that name.
```

@yujuhong @feiskyer

**Comment by yujuhong** [comment]
> 2) NPD failed to start because of device or resource busy. https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/1961
> This is a known issue, we should work around this.

Yes, we should detect this and work around it for the time being. 


> 1) e2e-image-puller was terminated by someone. https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gce-cri/1961

I didn't find this message in the log...

**Comment by Random-Liu** [comment]
@yujuhong Wrong paste.. Messed up by different links. I updated the comment just now.

**Comment by yujuhong** [comment]
> I didn't find anything weird in kubelet.log, but the container failed with a Terminated message. It seems that someone killed it, but not kubelet, because we can't see this log.

Docker got killed around that time:
```
time="2016-11-09T19:10:27.630777252Z" level=info msg="Processing signal 'terminated'"
time="2016-11-09T19:10:27.630882955Z" level=debug msg="starting clean shutdown of all containers..."
```

**Comment by yujuhong** [comment]
By the way, adding timestamps to `kern.log` in #36550.

**Comment by bprashanth** [comment]
Update on my previous comment (https://github.com/kubernetes/kubernetes/issues/33189#issuecomment-255901400), I'm starting to see more red herrings in production (b/32704760). Don't let a few instances of this line waylay you.

**Comment by feiskyer** [comment]
gci-gce-cri all green in recent tests, @yujuhong do we still have open failures?

**Comment by yujuhong** [comment]
We still have occasional flakes in the cluster e2e test. Let's keep this open for a few more days to see if my workaround would help.

**Comment by dims** [comment]
@yujuhong : can we close this now? If not, This needs to be triaged as a release-blocker or not for 1.5

**Comment by dims** [comment]
@yujuhong based on your last comment, i'll mark as non-release-blocker

**Comment by yujuhong** [comment]
I think we can close this one. Will file separate issues for test flakes.

**Comment by Random-Liu** [comment]
> I think we can close this one. Will file separate issues for test flakes.

SGTM
